DSharpPlus/Net/Rest/RateLimitStrategy.cs

```diff
@@ -152,40 +126,48 @@ public RateLimitPolicy(ILogger logger, int waitingForHashMilliseconds = 200)
             
             //We dont know the hash of the route, so we check if we have a request where we will get the hash. Otherwise
             //we will add the route to our list of routes we are waiting for a hash.
-            await this.waitingForHashListSemaphore.WaitAsync(cancellationToken);
+            await this.waitingForHashListSemaphore.WaitAsync(context.CancellationToken);
             if (this.waitingForHashRoutes.Contains(route))
             {
                 HttpResponseMessage synthesizedResponse = new(HttpStatusCode.TooManyRequests);
                 
-                DateTimeOffset reset = instant.AddMilliseconds(this.waitingForHashMilliseconds);
+                DateTimeOffset reset = instant.AddMilliseconds(waitingForHashMilliseconds);
 
                 synthesizedResponse.Headers.RetryAfter = new RetryConditionHeaderValue(reset);
                 synthesizedResponse.Headers.Add("DSharpPlus-Internal-Response", "waitingOnHash");
 
-                this.logger.LogWarning
+                logger.LogWarning
                 (
                     LoggerEvents.RatelimitPreemptive,
                     "Pre-emptive ratelimit triggered, waiting for route hash until {reset:yyyy-MM-dd HH:mm:ss zzz}.",
                     reset
                 );
                 waitingForHashRoutes.Add(route);
-                return synthesizedResponse;
+                return Outcome.FromResult(synthesizedResponse);
             }
             
             this.waitingForHashListSemaphore.Release();
         }
 
         // make the actual request
 
-        HttpResponseMessage response = await action(context, cancellationToken);
+        Outcome<HttpResponseMessage> outcome = await action(context, state);
+        
+        if (outcome.Result is null)
```

Comment:
I asume checking the result for null is the same as "is successful"?

Reply:
yes